diff --git a/ui/input-linux.c b/ui/input-linux.c
index 9720333b2c..b035399d54 100644
--- a/ui/input-linux.c
+++ b/ui/input-linux.c
@@ -146,6 +146,11 @@ static void input_linux_handle_keyboard(InputLinux *il,
          */
         if (il->grab_request && !il->keycount) {
             il->grab_request = false;
+            if (!il->grab_active) {
+                system("/home/xyz/software-kvm/swap.sh 1");
+            } else {
+                system("/home/xyz/software-kvm/swap.sh 0");
+            }
             input_linux_toggle_grab(il);
         }
     }
@@ -335,12 +340,12 @@ static void input_linux_complete(UserCreatable *uc, Error **errp)
     }
 
     qemu_set_fd_handler(il->fd, input_linux_event, NULL, il);
-    if (il->keycount) {
-        /* delay grab until all keys are released */
-        il->grab_request = true;
-    } else {
-        input_linux_toggle_grab(il);
-    }
+    // if (il->keycount) {
+    //     /* delay grab until all keys are released */
+    //     il->grab_request = true;
+    // } else {
+    //     input_linux_toggle_grab(il);
+    // }
     QTAILQ_INSERT_TAIL(&inputs, il, next);
     il->initialized = true;
     return;
